base/tuple.jl

```diff
@@ -615,3 +615,60 @@ empty(@nospecialize x::Tuple) = ()
 
 foreach(f, itr::Tuple) = foldl((_, x) -> (f(x); nothing), itr, init=nothing)
 foreach(f, itrs::Tuple...) = foldl((_, xs) -> (f(xs...); nothing), zip(itrs...), init=nothing)
+
+"""
+    circshift(x::Tuple, shift::Integer)
+
+Circularly shift, i.e. rotate, the data in a tuple. The second argument specifies
+the shift amount.
+
+# Examples
+```jldoctest
+julia> x = (1, 2, 3, 4, 5)
+(1, 2, 3, 4, 5)
+
+julia> circshift(x, 4)
+(2, 3, 4, 5, 1)
+
+julia> y = ((1, 2), (3, 4), (5, 6))
+((1, 2), (3, 4), (5, 6))
+
+julia> circshift(y, 5)
+((3, 4), (5, 6), (1, 2))
+
+julia> z = (1, 'a', -7.0, 3)
+(1, 'a', -7.0, 3)
+
+julia> circshift(z, -1)
+('a', -7.0, 3, 1)
+```
+"""
+function circshift(x::Tuple, shift::Integer)
+    j = mod1(shift, length(x))
+    ntuple(k -> __safe_getindex(x, k-j+ifelse(k>j,0,length(x))), Val(length(x)))
+end
+
+"""
+    circshift(x::Tuple, ::Val{shift})
```

Comment:
I would just merge these docs with the ones above by having a single docstring for `circshift(x::Tuple, shift)`

Reply:
Thank you again, @stevengj. This one's also done!